src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -374,6 +380,672 @@ static void appendModifiers(ImmutableArray<CustomModifier> customModifiers, Stri
+            {
+                return !typeParameter.ConstraintTypesNoUseSiteDiagnostics.IsEmpty || typeParameter.HasConstructorConstraint ||
+                    typeParameter.HasReferenceTypeConstraint || typeParameter.HasValueTypeConstraint ||
+                    typeParameter.HasNotNullConstraint || typeParameter.AllowsRefLikeType;
+            }
+
+            static void appendTypeWithAnnotation(TypeWithAnnotations type, StringBuilder builder)
+            {
+                appendType(type.Type, builder);
+
+                if (!type.Type.IsValueType)
+                {
+                    appendAnnotation(builder, type.NullableAnnotation);
+                }
+            }
+
+            static void appendAnnotation(StringBuilder builder, NullableAnnotation annotation)
+            {
+                switch (annotation)
+                {
+                    case NullableAnnotation.Annotated:
+                        builder.Append('?');
+                        break;
+                    case NullableAnnotation.NotAnnotated:
+                        builder.Append('!');
+                        break;
+                    case NullableAnnotation.Oblivious:
+                        break;
+                    case NullableAnnotation.Ignored:
+                        Debug.Assert(false);
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(annotation);
+                }
+            }
+
+            static void appendType(TypeSymbol type, StringBuilder builder)
+            {
+                if (type is NamedTypeSymbol namedType)
+                {
+                    appendNamedType(type, builder, namedType);
+                }
+                else if (type is TypeParameterSymbol { Name: var typeParameterName })
+                {
+                    appendIdentifier(typeParameterName, builder, forTypeConstraint: true);
+                }
+                else if (type is ArrayTypeSymbol array)
+                {
+                    appendArrayType(array, builder);
+                }
+                else if (type is PointerTypeSymbol pointer)
+                {
+                    appendTypeWithAnnotation(pointer.PointedAtTypeWithAnnotations, builder);
+                    builder.Append('*');
+                }
+                else if (type is FunctionPointerTypeSymbol functionPointer)
+                {
+                    appendFunctionPointerType(functionPointer, builder);
+                }
+                else if (type is DynamicTypeSymbol)
+                {
+                    builder.Append("dynamic");
+                }
+                else
+                {
+                    throw ExceptionUtilities.UnexpectedValue(type);
+                }
+            }
+
+            static void appendNamedType(TypeSymbol type, StringBuilder builder, NamedTypeSymbol namedType)
```

Comment:
>TypeSymbol type [](http://example.com/codeflow?start=40&length=15)

It looks like this parameter is redundant and is likely source of an inconsistent input for the function.